subsys/llext/llext.c

```diff
@@ -324,3 +324,189 @@ void llext_bootstrap(struct llext *ext, llext_entry_fn_t entry_fn, void *user_da
 		return;
 	}
 }
+
+/*
+ * Prepare a set of extension copies for future restoring. The user has copied
```

Comment:
If all extensions are in an array, and we need to save/restore to different memory potentially (while maintaining the links) can we not use array indices instead of pointers for this internally? Wouldn't that simply this entire thing?